### DISCUSSION TITLE
Add support for publishing sites to IPFS

### DISCUSSION THREAD

**Discussion by bluelovers** [vercel/community#80] [Idea]
hope support publish to ipfs like https://fleek.co/ do

1. publish to ipfs
2. support ipns

![image](https://user-images.githubusercontent.com/167966/81641839-8bb0b980-9454-11ea-9f76-04e78b28c41b.png)

![image](https://user-images.githubusercontent.com/167966/81642015-f661f500-9454-11ea-9881-213ee01af88a.png)

![image](https://user-images.githubusercontent.com/167966/81642026-fcf06c80-9454-11ea-9eeb-0ceb64319c08.png)

url style
- https://ipfs.fleek.co/ipfs/Qmd4ttxpV7sm3HvtyKzEBcwh34fwYmAt2qFhxFt1HroMZ7
- https://ipfs.fleek.co/ipns/ipfs-web-upload.on.fleek.co
- https://ipfs-web-upload.on.fleek.co/

https://github.com/zeit/now/issues/3649

**Comment by LatentM** [comment]
Yeah Vercel add support for ipfs. It's the future of web

**Comment by leerob** [comment]
We hear you and are looking into where this would fall on our roadmap 😄 

https://twitter.com/dApp_boi/status/1438046512093622276

**Comment by pseudozach** [comment]
not interested in ipfs but deploying to tor would be great!

**Reply by denzuko** [reply]
"Deploying to tor" means using git+ssh:// with torsocks and running a tor site in some linux box. Easier to just have a deployment branch and a cronjob pull that branch. But if one must go through extra steps then try pairing Vercel, ssh, and Pipedream. Doubt one is going to get tor with pipedream but one can try.

https://pipedream.com/apps/vercel-token-auth/integrations/ssh

Also check out how to host a onion site: https://medium.com/axon-technologies/hosting-anonymous-website-on-tor-network-3a82394d7a01

And here are some example steps to use git+ssh over tor:
```
# torsite side
cat - | crontab << EOL
@daily rm -rf /var/httpd/www && torsocks git clone -b production git+ssh://YourHash.onion:2222/FuBar/Site.git /var/httpd/www
EOL

**Comment by XahidEx** [comment]
Still waiting 🤗

**Comment by denzuko** [comment]
Mates, shouldn't be too hard to add a few nodes for Pinning then behind the scenes setting up some docker based pipelines. Slip into my DMs and my team at Da Planet Security can help your devs make this happen in six weeks.

**Comment by thekoenekamp** [comment]
+1

**Comment by outdoteth** [comment]
Any update on this?
